pkg/cmd/roachtest/tests/copyfrom.go

```diff
@@ -0,0 +1,162 @@
+CREATE INDEX  l_sk_pk ON lineitem (l_suppkey, l_partkey);
+`
+
+func initTest(ctx context.Context, t test.Test, c cluster.Cluster, sf int) {
+	if runtime.GOOS == "linux" {
+		if err := repeatRunE(
+			ctx, t, c, c.All(), "update apt-get", `sudo apt-get -qq update`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := repeatRunE(
+			ctx,
+			t,
+			c,
+			c.All(),
+			"install dependencies",
+			`sudo apt-get install -qq postgresql`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		csv := fmt.Sprintf(tpchLineitemFmt, sf)
+		c.Run(ctx, c.Node(1), "rm -f /tmp/lineitem-table.csv")
+		c.Run(ctx, c.Node(1), fmt.Sprintf("curl '%s' -o /tmp/lineitem-table.csv", csv))
+	}
+}
+
+func runTest(ctx context.Context, t test.Test, c cluster.Cluster, pg string) {
+	start := time.Now()
+	det, err := c.RunWithDetailsSingleNode(ctx, t.L(), c.Node(1), fmt.Sprintf(`cat /tmp/lineitem-table.csv | %s -c "COPY lineitem FROM STDIN WITH CSV DELIMITER '|';"`, pg))
+	if err != nil {
+		t.L().Printf("%v\n", det.Stdout)
+		t.L().Printf("%v\n", det.Stderr)
+		t.Fatal(err)
+	}
+	dur := time.Since(start)
+	t.L().Printf("%v\n", det.Stdout)
+	rows := 0
+	copy := ""
+	_, err = fmt.Sscan(det.Stdout, &copy, &rows)
+	require.NoError(t, err)
+	rate := int(float64(rows) / dur.Seconds())
+	t.L().Printf("results: %d rows/s", rate)
+	// Write the copy rate into the stats.json file to be used by roachperf.
+	c.Run(ctx, c.Node(1), "mkdir", t.PerfArtifactsDir())
+	cmd := fmt.Sprintf(
+		`echo '{ "copy_rate": %d }' > %s/stats.json`,
+		rate, t.PerfArtifactsDir(),
+	)
+	c.Run(ctx, c.Node(1), cmd)
+}
+
+func runCopyFromPG(ctx context.Context, t test.Test, c cluster.Cluster, sf int) {
+	initTest(ctx, t, c, sf)
+	c.Run(ctx, c.Node(1), "sudo -i -u postgres psql -c 'DROP TABLE IF EXISTS lineitem'")
+	c.Run(ctx, c.Node(1), fmt.Sprintf("sudo -i -u postgres psql -c '%s'", lineitemSchema))
+	runTest(ctx, t, c, "sudo -i -u postgres psql")
+}
+
+func runCopyFromCRDB(ctx context.Context, t test.Test, c cluster.Cluster, sf int) {
+	c.Put(ctx, t.Cockroach(), "./cockroach", c.All())
+	c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings(), c.All())
+	initTest(ctx, t, c, sf)
+	urls, err := c.InternalPGUrl(ctx, t.L(), c.Node(1))
+	require.NoError(t, err)
+	m := c.NewMonitor(ctx, c.All())
+	m.Go(func(ctx context.Context) error {
+		// psql w/ url first are doesn't support --db arg so have to do this.
+		url := strings.Replace(urls[0], "?", "/defaultdb?", 1)
+		c.Run(ctx, c.Node(1), fmt.Sprintf("psql %s -c 'SELECT 1'", url))
+		c.Run(ctx, c.Node(1), fmt.Sprintf("psql %s -c '%s'", url, lineitemSchema))
+		runTest(ctx, t, c, fmt.Sprintf("psql '%s'", url))
+		return nil
+	})
+	m.Wait()
+}
+
+func registerCopyFrom(r registry.Registry) {
+	testcases := []struct {
+		sf    int
+		nodes int
+	}{
+		{sf: 1, nodes: 1},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
+		r.Add(registry.TestSpec{
+			Name:    fmt.Sprintf("copyfrom/crdb/sf=%d,nodes=%d", tc.sf, tc.nodes),
```

Comment:
should we use `sf=%d/nodes=%d`? haven't seen commas much.

Reply:
👍